psutil/disk.go

```diff
@@ -0,0 +1,153 @@
+///*
+//http://www.apache.org/licenses/LICENSE-2.0.txt
+//
+//Copyright 2017 Intel Corporation
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+//*/
+//
+package psutil
+
+import (
+	"strings"
+	"time"
+
+	"github.com/shirou/gopsutil/disk"
+
+	"github.com/intelsdi-x/snap/control/plugin"
+	"github.com/intelsdi-x/snap/core"
+)
+
+func getPSUtilDiskUsage(path string) (*disk.UsageStat, error) {
+	disk_usage, err := disk.Usage(path)
+	if err != nil {
+		return nil, err
+	}
+	return disk_usage, nil
+}
+
+func getDiskUsageMetrics(nss []core.Namespace, mounts []string) ([]plugin.MetricType, error) {
+	t := time.Now()
+	var paths []disk.PartitionStat
+	metrics := []plugin.MetricType{}
+	namespaces := map[string][]string{}
+	requested := map[string]core.Namespace{}
+	for _, ns := range nss {
+		namespaces[ns.Strings()[len(ns.Strings())-1]] = ns.Strings()
+		requested[ns.Strings()[len(ns.Strings())-1]] = ns
+	}
+	if strings.Contains(mounts[0], "all") {
+		paths = make([]disk.PartitionStat, 0)
+		parts, err := disk.Partitions(false)
+		if err != nil {
+			return nil, err
+		}
+		for _, part := range parts {
```

Comment:
If there is a reason to avoid simple assignment `paths = parts` you could use `append(paths, parts...)`, but I bet simple assignment there is enough.